internal/tracing/handlers/handler.go

```diff
@@ -57,3 +59,78 @@ func BuildOptions(k8s client.Client, mcAddon *addonapiv1alpha1.ManagedClusterAdd
 
 	return resources, nil
 }
+
+func getSecretExporters(otelCol *otelv1alpha1.OpenTelemetryCollector) (map[authentication.Target]string, error) {
+	exporterSecrets := map[authentication.Target]string{}
+
+	exporters, err := getExporters(otelCol)
+	if err != nil {
+		return exporterSecrets, err
+	}
+
+	if len(exporters) == 0 {
+		return exporterSecrets, errNoExportersFound
+	}
+
+	for _, vol := range otelCol.Spec.Volumes {
+		// We only care about volumes created from secrets
+		if vol.Secret != nil {
+			vm, err := getVolumeMount(otelCol, vol.Secret.SecretName)
+			if err != nil {
+				return exporterSecrets, err
+			}
+			exporter, err := searchVolumeMountInExporter(vm, exporters)
+			if err != nil {
+				klog.Warning(err)
+				continue
```

Comment:
can a secret be mounted anywhere else aside from exporters? If so does it also make sense to print a warning?

Reply:
It could be used for the receivers but it is not as usual. I would keep it and add extra checks if needed in the future.